mythril/laser/ethereum/state/world_state.py

```diff
@@ -73,26 +73,25 @@ def __copy__(self) -> "WorldState":
         new_world_state.constraints = copy(self.constraints)
         return new_world_state
 
-    def accounts_exist_or_load(self, addr: str, dynamic_loader: DynLoader) -> str:
+    def accounts_exist_or_load(self, addr: str, dynamic_loader: DynLoader) -> Account:
         """
         returns account if it exists, else it loads from the dynamic loader
         :param addr: address
         :param dynamic_loader: Dynamic Loader
         :return: The code
         """
         addr_bitvec = symbol_factory.BitVecVal(int(addr, 16), 256)
+
         if addr_bitvec.value in self.accounts:
-            code = self.accounts[addr_bitvec.value].code
-        else:
-            code = dynamic_loader.dynld(addr)
-            self.create_account(
-                balance=0, address=addr_bitvec.value, dynamic_loader=dynamic_loader
-            )
-        if code is None:
-            code = ""
-        else:
-            code = code.bytecode
-        return code
+            return self.accounts[addr_bitvec.value]
+        if dynamic_loader is None:
+            raise ValueError("dynamic_loader is None")
+        return self.create_account(
+            balance=0,
```

Comment:
Should we not take the concrete balance from on-chain?

Reply:
yes, Let's add it after this PR.

Reply:
👍